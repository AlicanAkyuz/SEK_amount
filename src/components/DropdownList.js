import React from 'react';
import PropTypes from 'prop-types';

import { ListItem, PrimaryText, ListText } from '../styles';

const DropdownList = ({
  name, capital, population, currencyCode, sekValue,
}) => (
  <ListItem>
    <PrimaryText>{name}</PrimaryText>
    <ListText>{`Capital City: ${capital}`}</ListText>
    <ListText>{`Population: ${population}`}</ListText>
    <ListText>{`Currency: ${currencyCode}`}</ListText>
    <ListText>{sekValue}</ListText>
  </ListItem>
);

DropdownList.propTypes = {
  name: PropTypes.string.isRequired,
  capital: PropTypes.string.isRequired,
  population: PropTypes.string.isRequired,
  currencyCode: PropTypes.string.isRequired,
  sekValue: PropTypes.string.isRequired,
};

export default DropdownList;
