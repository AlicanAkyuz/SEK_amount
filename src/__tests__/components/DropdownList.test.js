import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DropdownList from '../../components/DropdownList';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <DropdownList
      name="Germany"
      capital="Berlin"
      population="80000000"
      currencyCode="EUR"
      sekValue="0.98"
    />,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('correctly renders its required props as defined by prop types', () => {
  const tree = renderer
    .create(
      <DropdownList
        name="Germany"
        capital="Berlin"
        population="80000000"
        currencyCode="EUR"
        sekValue="0.98"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
