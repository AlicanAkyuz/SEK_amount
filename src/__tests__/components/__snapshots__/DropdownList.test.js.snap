// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders its required props as defined by prop types 1`] = `
<div
  className="css-bp3e54"
>
  <h3
    className="css-qwi5oq"
  >
    Germany
  </h3>
  <h3
    className="css-f71r11"
  >
    Capital City: Berlin
  </h3>
  <h3
    className="css-f71r11"
  >
    Population: 80000000
  </h3>
  <h3
    className="css-f71r11"
  >
    Currency: EUR
  </h3>
  <h3
    className="css-f71r11"
  >
    0.98
  </h3>
</div>
`;
